app/auth.go

```diff
@@ -356,17 +356,23 @@ func (m *menderAuthManagerService) broadcast(message AuthManagerResponse) {
 		}
 	}
 	// emit signal on dbus, if available
-	if m.dbus != nil {
+	if m.dbus != nil && message.AuthToken != noAuthToken {
 		m.dbus.EmitSignal(m.dbusConn, "", AuthManagerDBusPath,
 			AuthManagerDBusInterfaceName, AuthManagerDBusSignalValidJwtTokenAvailable)
+		log.Debugf("dbus: emitting signal: %s", AuthManagerDBusSignalValidJwtTokenAvailable)
 	}
 }
 
 // broadcastAuthTokenAvailable broadcasts the notification to all the subscribers
 func (m *menderAuthManagerService) broadcastAuthTokenAvailable() {
 	authToken, err := m.authToken()
+	log.Debugf("broadcastAuthTokenAvailable: running after m.authToken:%s err:%v", authToken, err)
 	if err == nil && authToken != noAuthToken {
-		m.broadcast(AuthManagerResponse{Event: EventAuthTokenAvailable})
+		m.broadcast(AuthManagerResponse{
+			AuthToken: authToken,
+			Event:     EventAuthTokenAvailable,
+			Error:     nil,
+		})
```

Comment:
@kacf @tranchitella @lluiscampos so I am dropping this part, and keep the rest, right? (after the discussion a couple of minutes ago
)

Reply:
If we do [MEN-4191](https://tracker.mender.io/browse/MEN-4191), then the original is correct, we should always emit the signal. So I think this PR is not needed anymore.